Code/Framework/AzToolsFramework/AzToolsFramework/API/PythonLoader.h

```diff
@@ -18,8 +22,46 @@ namespace AzToolsFramework::EmbeddedPython
         PythonLoader();
         ~PythonLoader();
 
+        //! Calculate the python home (PYTHONHOME) based on the engine root
+        //! @param engineRoot The path to the engine root to locate the python home
+        //! @return The path of the python home path
+        static AZ::IO::FixedMaxPath GetPythonHomePath(AZStd::string_view engineRoot);
+
+        //! Collect the paths from all the egg-link files found in the python home
+        //! paths used by the engine
+        //! @param thirdPartyRoot The root location of the O3DE 3rdParty folder
+        //! @param engineRoot The path to the engine root to locate the python home
+        //! @param resultPaths The list of paths to discovered when searching through python home
+        static void ReadPythonEggLinkPaths(AZ::IO::PathView thirdPartyRoot, AZStd::string_view engineRoot, AZStd::vector<AZStd::string>& resultPaths);
```

Comment:
I recommend making the `resultPath` parameter a vector of `AZ::IO::Path` instead of AZStd::string.

Also the engineRoot parameter type(`AZStd::string_view`) is inconsistent with the `thirdPartyRoot` type(`AZ::IO::PathView`)

It would help avoid mistakes for the caller when attempting to compare the path later on in case there are slash ('/' vs '\') mismatches or on Windows different cases.
```suggestion
        static void ReadPythonEggLinkPaths(AZ::IO::PathView thirdPartyRoot, AZ::IO::PathView engineRoot, AZStd::vector<AZ::IO::Path>& resultPaths);
```

Another approach is to have the `ReadPythonEggLinkPaths` function accept a callback function, that is invoked every time an egg-link path is found.
i.e
```c++
using EggLinkPathVisitor = AZStd::function<void(AZ::IO::PathView)>;
static void VisitPythonEggLinkPaths(AZ::IO::PathView thirdPartyRoot, AZStd::string_view engineRoot, EggLinkPathVisitor eggLinkVisitor);
```

With that the ReadPythonEggLinkPaths function would then only need to call the VisitPythonEggLinkPaths function to populate it's vector.
```c++
void ReadPythonEggLinkPaths(AZ::IO::PathView thirdPartyRoot, AZStd::string_view engineRoot, AZStd::vector<AZ::IO::Path>& resultPaths)
{
    auto AppendEggLinks = [&resultPaths](AZ::IO::PathView eggLinkPath)
    {
        resultPaths.emplace_back(eggLinkPath);
    };
    VisitPythonEggLinkPaths(thirdPartyRoot, engineRoot, AZStd::move(AppendEggLinks));
}
```

Reply:
I'm going to ahead with the visitor approach, because there is code in EditorPythonBindings that uses [PythonPathStack](https://github.com/spham-amzn/o3de/blob/e7e6d832d90dcddc81a5e9c3c031a633cbddd81f/Gems/EditorPythonBindings/Code/Source/PythonSystemComponent.h#L91), which is a already a vector of strings.